### PR TITLE
ci: adding rocm 5.7 support

### PR DESCRIPTION
**Pull request by romintomasetti**
This PR adds `rocm-5.7` to the pipeline.

Note that at the beginning, I tried with `rocm-5.6`, but got weird issues.

### CODE REVIEW

**Comment by romintomasetti** [comment]
Hi @dalg24 I wanted to give the CI a fresh look, but it seems some nasty undetected bug appeared... Any idea?

**Comment by dalg24** [comment]
I would do a make verbose and look at the compile line.  It is nonsensical, going through Cang CUDA wrappers header files.

**Comment by romintomasetti** [comment]
> I would do a make verbose and look at the compile line. It is nonsensical, going through Cang CUDA wrappers header files.

Done [in this job](https://github.com/kokkos/kokkos-tools/actions/runs/6085098459/job/16508407567). I don't see any additional helping piece of info coming in...

BTW, the issue could be related to https://github.com/ROCm-Developer-Tools/HIP/issues/2082.

**Comment by romintomasetti** [comment]
@dalg24 It seems that with `rocm-5.7` everything works!

If you agree, I would proceed with this PR :smile:

**Review by masterleinad — APPROVED** [review]
Fine with me. Please update the pull request title and description to reflect the most recent changes.

**Comment by romintomasetti** [comment]
Hi @dalg24 @masterleinad ! Would you have a second to approve and merge? Thanks!

**Review by masterleinad — APPROVED** [review]

**Review by vlkale — APPROVED** [review]

**Comment by romintomasetti** [comment]
@dalg24 It seems we still need one approval of someone with write access :wink:

**Comment by romintomasetti** [comment]
@vlkale Who can additionally approve this one? (with write access)

**Comment by vlkale** [comment]
> @vlkale Who can additionally approve this one? (with write access)


@crtrott or @masterleinad: Can you please look at this when possible?

File: .github/workflows/build-with-kokkos.yml
Comment (dalg24) [inline .github/workflows/build-with-kokkos.yml]: Did you mean to commit that change?

Comment (romintomasetti) [reply .github/workflows/build-with-kokkos.yml]: Indeed, no. I reverted that change :heavy_check_mark:

**Pull request merged**
